tracer/src/Datadog.Trace.Tools.Runner/RunCommand.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            var args = settings.Command ?? context.Remaining.Raw;
+
+            if (args.Count == 0)
+            {
+                return ValidationResult.Error("Missing command");
+            }
+
+            if (settings.AdditionalEnvironmentVariables != null)
+            {
+                foreach (var env in settings.AdditionalEnvironmentVariables)
+                {
+                    if (!env.Contains('='))
+                    {
+                        return ValidationResult.Error($"Badly formatted environment variable: {env}");
+                    }
+                }
+            }
+
+            return base.Validate(context, settings);
+        }
+
+        protected int Execute(CommandContext context, RunSettings settings, bool enableCiMode)
+        {
+            var args = settings.Command ?? context.Remaining.Raw;
+
+            var profilerEnvironmentVariables = Utils.GetProfilerEnvironmentVariables(
+                ApplicationContext.RunnerFolder,
+                ApplicationContext.Platform,
+                settings);
+
+            if (profilerEnvironmentVariables is null)
+            {
+                return 1;
+            }
+
+            if (enableCiMode)
+            {
+                profilerEnvironmentVariables[Configuration.ConfigurationKeys.CIVisibilityEnabled] = "1";
+            }
+
+            if (settings.AdditionalEnvironmentVariables != null)
+            {
+                foreach (var env in settings.AdditionalEnvironmentVariables)
+                {
+                    var (key, value) = ParseEnvironmentVariable(env);
+
+                    profilerEnvironmentVariables[key] = value;
+                }
+            }
+
+            var commandLine = string.Join(' ', args);
+
+            if (!string.IsNullOrWhiteSpace(commandLine))
+            {
+                // CI Visibility mode is enabled we check if we have connection to the agent before running the process.
+                if (enableCiMode && !Utils.CheckAgentConnectionAsync(settings.AgentUrl).GetAwaiter().GetResult())
+                {
+                    return 1;
+                }
+
+                AnsiConsole.WriteLine("Running: " + commandLine);
+
+                var arguments = args.Count > 1 ? string.Join(' ', args.Skip(1).ToArray()) : null;
+
+                if (Program.CallbackForTests != null)
+                {
+                    Program.CallbackForTests(args[0], arguments, profilerEnvironmentVariables);
+                    return 0;
+                }
+
+                var processInfo = Utils.GetProcessStartInfo(args[0], Environment.CurrentDirectory, profilerEnvironmentVariables);
+
+                if (args.Count > 1)
+                {
+                    processInfo.Arguments = arguments;
+                }
+
+                return Utils.RunProcess(processInfo, ApplicationContext.TokenSource.Token);
+            }
+
+            return 0;
```

Comment:
If I'm following correctly, we'll hit this if we run `dd-trace run`? If so, then isn't that an incorrect usage? Or maybe this can never be hit due to validation/command configuration? Either way, I think it should return `1` at least, and potentially a comment why it can/can't be reached?

Reply:
Good catch. It should be unreachable because of the validation, but it should definitely return 1

Reply:
Actually I'm going to remove this state entirely and add a test to make sure it's covered by validation

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2295/commits/54c8e313695aae194ab141e73e2b033a62f3f06b